common/current_preset.h

```diff
@@ -271,10 +271,30 @@ class CurrentPreset {
 
   void SetPreset(int preset) {
     Clear();
+    #ifdef SAVED_PRESET
+    SavePreset(preset);
+    #endif
     LOCK_SD(true);
     if (!Load(preset)) Set(preset);
     LOCK_SD(false);
   }
+
+  bool SavePreset(int preset) {
+    if (preset >= 0 && preset <= current_config->num_presets) {
+      STDOUT.println("Saving Current Preset");
+      LOCK_SD(true);
+      FileReader c;
+      LSFS::Remove("savedpreset.ini");
+      c.Create("savedpreset.ini");
+      char value[30];
+      itoa(preset, value, 10);
+      c.write_key_value("preset", value);
```

Comment:
In the TD code, this is actually done with "rotate_presets", which simply re-orders the presets in presets.ini and saves it.
Not sure which way is better, but I wanted to point out the option at least.